python-frontend/src/main/java/org/sonar/python/types/InferredTypes.java

```diff
@@ -102,8 +102,9 @@ public class InferredTypes {
   private static final Set<Set<String>> HARDCODED_INCOMPATIBLE_TYPES = Set.of(Set.of(BuiltinTypes.TUPLE, BuiltinTypes.LIST));
 
   static {
-    HARDCODED_COMPATIBLE_TYPES.put(BuiltinTypes.INT, new HashSet<>(Arrays.asList(BuiltinTypes.FLOAT, BuiltinTypes.COMPLEX)));
-    HARDCODED_COMPATIBLE_TYPES.put(BuiltinTypes.FLOAT, new HashSet<>(Collections.singletonList(BuiltinTypes.COMPLEX)));
+    HARDCODED_COMPATIBLE_TYPES.put(BuiltinTypes.INT, new HashSet<>(Arrays.asList(BuiltinTypes.FLOAT, BuiltinTypes.COMPLEX, BuiltinTypes.BOOL)));
```

Comment:
I think `BuiltinTypes.BOOL` is not needed here. Similarly to the fact we raise an issue when `float` is provided to a function expecting `bool`, it would make sense to raise an issue when passing `int` to a function expecting `bool` (Mypy raises the issue).

In this case, for some unrelated reason, the issue is not raised (actually because `int` defines all the members that `bool` defines and so our generic duck-type compatibility check returns true - and it's okay), but there is no need to hard-code the fact that `int` can be provided where `bool` is expected. (It would probably make sense to have another test highlighting the behavior though: `takes_boolean(42) # FN`).